apps/frontend/static_src/js/main.js

```diff
@@ -1 +1,70 @@
 import { Dropdown } from 'bootstrap';
+
+const searchInput = document.querySelector('[data-search-input]')
+const searchIconButton = document.querySelector('[data-search-icon-button]')
+
+const onSearchInputChange = async (event) => {
+    const query = event.target.value
+    try {
+        const res = await fetch(window.location.origin + '/search_json/?query=' + query)
+        const data = await res.json()
+        injectResultsInHTML(data)
+    } catch(err) {
+        console.log(err)
+        return {"error": err}
+    }
+}
+searchInput.addEventListener("keyup", (event) => {
+    setTimeout(() => {
+        onSearchInputChange(event)
+    }, 1000)
+})
```

Comment:
This seems to only delay the `fetch()` without preventing duplicate requests:

<img width="1310" alt="image" src="https://user-images.githubusercontent.com/6379424/184891527-6148fa34-f14e-4682-a42e-94352d6fd571.png">

Are you familiar with [debouncing/throttling](https://css-tricks.com/debouncing-throttling-explained-examples/)? I think that could help eliminate the duplicate requests.

Reply:
Nope I am not familiar with that. I will read about that and will try to implement it.